apollo-federation/src/sources/graphql/federated_query_graph/builder.rs

```diff
@@ -133,7 +133,7 @@ fn process_type(
             .filter(|field| is_field_ignored(field.clone().into(), subgraph_schema))
             .try_for_each(|subgraph_field| {
                 subgraph_schema
-                    .get_type(subgraph_field.field_name.clone())
+                    .get_type(subgraph_field.type_name.clone())
```

Comment:
is the type name the parent type (`Type` in `Type.field(): Output`) or the output type (`Output` in `Type.field(): Output`)?

Reply:
It should be the return type of the field.

Reply:
maybe i'm missing something, but i don't think that's true: https://github.com/apollographql/router/blob/dev/apollo-federation/src/schema/position.rs#L1776-L1782 — the type_name comes from the parent type.

Reply:
ohhhh sorry, thanks for the clarification!

Reply:
To make sure that I have the right idea, I need to replace `subgraph_field.type_name.clone()` with `subgraph_field.get(subgraph_schema.schema())?.ty.inner_named_type()` in order to get the correct type name. Put together, this would be get the inner type name then call `subgraph_schema.get_type(inner_type_name)`, i.e.
```rust
// Old
subgraph_schema
    .get_type(subgraph_field.type_name.clone())
// New
subgraph_field
    .get(subgraph_schema.schema())
    .and_then(|field| subgraph_schema.get_type(field.ty.inner_named_type().clone()))
```

Reply:
@TylerBloom Yep, that looks right to me